### PR TITLE
[PW 1949]: Notifications error_count and error_message

### PR DESCRIPTION
**Pull request by acampos1916**
**Description**
Added an error count and error message columns to adyen_notification table and Notifications Overview. UpgradeSchema has been modified for this.

**Tested scenarios**
Placed and order, received notification and it now shows in Notifications Overview. Thrown an exception during the notification processing and the exception is appended to error_message, error_count increased by 1.

**Fixed issue**:  PW-1949

### CODE REVIEW

File: etc/module.xml [hunk 1]
```diff
@@ -24,7 +24,7 @@
 -->
 <config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Module/etc/module.xsd">
 
-    <module name="Adyen_Payment" setup_version="4.5.4">
+    <module name="Adyen_Payment" setup_version="5.3.1">
```
Comment (acampos1916) [inline etc/module.xml]: Is this the correct usage of setup_version according to our release convention? https://hub.is.adyen.com/our-company/organization/workstreams/mid-market-solution/plugins-and-partners/internal-repo/release-steps/magento2/#release-on-github

File: Model/Notification.php [hunk 2]
```diff
@@ -49,6 +49,7 @@ class Notification extends \Magento\Framework\Model\AbstractModel implements Not
     const REPORT_AVAILABLE = "REPORT_AVAILABLE";
     const ORDER_CLOSED = "ORDER_CLOSED";
     const OFFER_CLOSED = "OFFER_CLOSED";
+    const MAX_ERROR_COUNT = 5;
```
Comment (acampos1916) [inline Model/Notification.php]: Hard limit for now

File: Model/Notification.php [hunk 3]
```diff
@@ -359,6 +360,69 @@ public function setDone($done)
...
+     * Sets Processing flag.
+     *
+     * @param bool $processing
+     * @return $this
+     */
+    public function setProcessing($processing)
+    {
+        return $this->setData(self::PROCESSING, $processing);
+    }
+
```
Comment (acampos1916) [inline Model/Notification.php]: These getter and setter for processing are not needed for this PR but I noticed that they were missing

File: Model/Notification.php [hunk 4]
```diff
@@ -359,6 +360,69 @@ public function setDone($done)
...
+    /**
+     * Sets Processing flag.
+     *
+     * @param bool $processing
+     * @return $this
+     */
+    public function setProcessing($processing)
+    {
+        return $this->setData(self::PROCESSING, $processing);
+    }
```
Comment (acampos1916) [inline Model/Notification.php]: These getter and setter for processing are not needed for this PR but I noticed that they were missing

File: etc/module.xml (hunk 1 above)
Comment (msilvagarcia) [reply etc/module.xml]: Yes, I believe so.

Comment (cyattilakiss) [reply etc/module.xml]: it's gonna be a feature release so I would go for ->

```suggestion
    <module name="Adyen_Payment" setup_version="5.4.0">
```

**Pull request merged**